packages/react-is/src/__tests__/ReactIs-test.js

```diff
@@ -106,13 +108,15 @@ describe('ReactIs', () => {
 
   it('should identify ref forwarding component', () => {
     const RefForwardingComponent = React.forwardRef((props, ref) => null);
+    expect(ReactIs.isValidElementType(RefForwardingComponent)).toBe(true);
     expect(ReactIs.typeOf(<RefForwardingComponent />)).toBe(ReactIs.ForwardRef);
```

Comment:
I noticed an inconsistency here with how typeOf works when we pass a momozied element vs when we pass a ref-forwarding element. More info here https://github.com/facebook/react/pull/17274#discussion_r342368003

I think this PR addresses this inconsistency between the two types, but I'm still curious, what's the intended behavior for `typeOf(RefForwardingComponent)` and `typeOf(<RefForwardingComponent />)` – should we expect to get back `ReactIs.ForwardRef` in both cases? Same question goes for memoized elements/components.

For example:

```js
expect(ReactIs.typeOf(<Memoized />)).toBe(ReactIs.Memo);
expect(ReactIs.typeOf(Memoized)).toBe(ReactIs.Memo);

expect(ReactIs.typeOf(<RefForwardingComponent />)).toBe(ReactIs.ForwardRef);
expect(ReactIs.typeOf(RefForwardingComponent)).toBe(ReactIs.ForwardRef);
```

The proposed bug fix in #17274 relies on those assertions above being valid.

Reply:
> I think this PR addresses this inconsistency between the two types, but I'm still curious, what's the intended behavior for typeOf(RefForwardingComponent) and typeOf(<RefForwardingComponent />) – should we expect to get back ReactIs.ForwardRef in both cases?

No. That's the primary purpose of this PR, to _not_ support that use case.

I've seen #17274. I have some draft comments on it but got distracted by this PR first.

Reply:
Thanks for the clarification @bvaughn! Exactly the answer I was looking for!